docs/shared/using-nx/affected.md

```diff
@@ -1,27 +1,122 @@
-# How Affected Works
+# Run Only Tasks Affected by a PR
```

Comment:
Overall for this one. Can we also make the headings more descriptive? 

Right now they are not really useful
![image](https://github.com/nrwl/nx/assets/542458/e4baf58c-2382-413a-81bd-2ee99a853ea9)

Maybe something like

- How Affected Works
- Visualizing Affected Projects (with the Graph)
- Determining the base SHA to compare with (or something similar)
- Ignoring Files from Affected Commands
- Not using Git

Reply:
Some more generic comments because I cannot comment directly on the line since it didn't change in this PR 🤷‍♂️.

- In the current "CI" section I'd also link to the various ways we have to find the last successful SHA etc.
- In the "Not Using Git" I'd mention that Nx uses Git to determine the affected files, otherwise it might be confusing. Or maybe we should mention that in the "How it works" section. I don't think we did that.

Reply:
Each CI provider has a different way of finding the last successful SHA, so I put in a generic statement about the importance of comparing to the last successful commit.

Reply:
It would probably be worth linking the [nx-orb](https://github.com/nrwl/nx-orb) and [set-shas](https://github.com/nrwl/nx-set-shas) right?

Reply:
ok, I can mention them.  They're specific to Circle CI and Github Actions though, so people who aren't using those CI providers won't know what to do